lib/puppet/property/list.rb

```diff
@@ -7,19 +7,6 @@ class Property
     #
     class List < Property
 
-      def should_to_s(should_value)
-        #just return the should value
-        should_value
-      end
-
-      def is_to_s(currentvalue)
-        if currentvalue == :absent
-          return _("absent")
-        else
-          return currentvalue.join(delimiter)
```

Comment:
The `delimiter` can and is overridden for specific types. I don't think `format_value_for_display` takes that into account.